spring-amqp/src/main/java/org/springframework/amqp/core/Address.java

```diff
@@ -39,14 +39,25 @@
  */
 public class Address {
 
+	/**
+	 * Use this value in {@code RabbitTemplate#setReplyAddress(String)} to use a temporary reply queue
+	 * for each request, regardless of whether the broker supports direct reply-to.
+	 */
+	public static final String TEMPORARY_REPLY_QUEUE_TOKEN = "##TEMPORARY_QUEUE##";
```

Comment:
I agree with Artem - Let's do it in 1.6 in a clean way so everybody will be happy ;)
Gary, I need such option because I am making RPC call which is processed by some non java app. This app expects temporary queue. Because such queue doesn't exist it throws exception and do not work correctly - with temporary queue everything works. I agree that when you have control over client and server then direct reply-to is the best, efficient solution.